### PR TITLE
Miscellaneous docstring edits

### PR DESCRIPTION
**Pull request by djmitche**
There are a few semantic changes here, based on the code; please verify?

### CODE REVIEW

File: routes/api/v1.js [hunk 1]
```diff
@@ -142,13 +142,12 @@ api.declare({
...
-    "Create client with given `clientId`, `name`, `expires`, `scopes` and",
+    "Create a client with given `clientId`, `name`, `expires`, `scopes` and",
     "`description`. The `accessToken` will always be generated server-side,",
     "and will be returned from this request.",
     "",
-    "**Required scopes**, in addition the scopes listed",
-    "above, the caller must also posses the all the scopes that is given to",
-    "the client that is created."
+    "**Required scopes**: in addition the scopes listed",
+    "above, the `scopes` parameter must satisfy the caller's scopes."
```
Comment (jonasfj) [inline routes/api/v1.js]: I think "the `scopes` parameter must satisfy the caller's scopes" got flipped around.

```
  "**Required scopes**: in addition the scopes listed",
  "above, the caller must satisfy all scopes granted to the client."
```
or if you must:
```
  "**Required scopes**: in addition the scopes listed",
  "above, the caller must satisfy all scopes from the `scopes` property of the payload."
```

File: routes/api/v1.js [hunk 2]
```diff
@@ -223,9 +222,9 @@ api.declare({
     "Modify client `name`, `expires`, `scopes` and",
     "`description`.",
     "",
-    "**Required scopes**, in addition the scopes listed",
-    "above, the caller must also posses the all the scopes that is given to",
-    "the client that is updated."
+    "**Required scopes**: in addition the scopes listed",
+    "above, the `scopes` parameter must satisfy the caller's scopes.  The",
+    "client's existing scopes need not satisfy the caller's scopes."
```
Comment (jonasfj) [inline routes/api/v1.js]: Again this is reversed...
It's the caller that must satisfy all the scopes assigned to the modified client. Also I'm not happy calling it the `scopes` parameter, it's the `scopes` property or `client.scopes` property, or `scopes` property of the payload. That's mostly nit...

However, direction of satisfiability is important.

Comment (jonasfj) [inline routes/api/v1.js]: note: `client.scopes <= caller.scopes` where `client` is the payload. Hence, caller must satisfy the scopes granted to the client.

File: routes/api/v1.js [hunk 3]
```diff
@@ -439,20 +440,20 @@ api.declare({
...
     "application.",
     "",
-    "Please notice that your `prefix` may not start with slash `/`, it is",
-    "allowed on S3, but we forbid it here to discourage bad behavior.",
-    "Also note that if your `prefix` doesn't end in a slash `/` the STS",
-    "credentials will not require one to be to inserted. This is mainly a",
+    "Please note that your `prefix` may not start with slash `/`. Such a prefix",
+    "is allowed on S3, but we forbid it here to discourage bad behavior.",
+    "Also note that if your `prefix` doesn't end in a slash `/`, the STS",
+    "credentials will not require a trailing slash. This is mainly a",
```
Comment (jonasfj) [inline routes/api/v1.js]: Trailing slash is a different thing.
You prefix can be `my-folder/` in which case you can modify `my-folder/my-file.txt`, etc...

If the prefix doesn't have a trailing prefix, one will not be required to be inserted, and hence, if you have prefix `my-folder` you may modify `my-folder/my-file.txt` as before, but you may also modify `my-folder-NOT.txt` :)

So if you security relies on folder structures and not prefixes one must include trailing slash in the prefix.

File: routes/api/v1.js (hunk 1 above)
Comment (djmitche) [inline routes/api/v1.js]: Yeah, I posted this before we talked in irc.  I'll revise.

File: routes/api/v1.js (hunk 3 above)
Comment (djmitche) [inline routes/api/v1.js]: I think we're saying the same thing - "one will not be required to be inserted" is a double-passive form of "there is no need to insert one", where "one" refers to a trailing slash.  But I'll try to say it a different way.

**Comment by djmitche** [comment]
OK, updated -- have another look?  Sorry again for my confusion!

File: routes/api/v1.js [hunk 4]
```diff
@@ -223,9 +222,9 @@ api.declare({
     "Modify client `name`, `expires`, `scopes` and",
     "`description`.",
     "",
-    "**Required scopes**, in addition the scopes listed",
-    "above, the caller must also posses the all the scopes that is given to",
-    "the client that is updated."
+    "**Required scopes**: in addition the scopes listed",
+    "above, the `scopes` property must be satisfied by the caller's",
+    "scopes.  The client's existing scopes are not considered."
```
Comment (jonasfj) [inline routes/api/v1.js]: This is nit, but if we do have colon, I believe one should follow with a capital "In", not sure though... Either way, I was happy with comma and remains happy with colon :)

Note, `satisfied by the caller` is likely sufficient to imply satisfied by the caller's scopes. But again, it's nit picking...

**Pull request merged**